### DISCUSSION TITLE
circular physics bodies?

### DISCUSSION THREAD

**Discussion by adamwong246** [KilledByAPixel/LittleJS#91] [General]
Phaser has this: https://newdocs.phaser.io/docs/3.80.0/focus/Phaser.Physics.Arcade.Image-setCircle

Can littleJs provide such a feature?

**Comment by KilledByAPixel** [comment]
This is the kind of thing that will get automatically with a Box2d plugin which will support all kinds of shapes. I hope to work on this soon.

The built in arcade physics implementation for LittleJS is really solid and simple but only for axis aligned boxes. I'm not sure it can easily be converted to handle circles. You can test the distance between the objects in collision functions to make them behave more like circles in some cases, like for a large circular shaped projectile that you want to have special collision for.

**Comment by KilledByAPixel** [comment]
Box2D is now included with LittleJS, and supports circular physics bodies!

![screenshot (28)](https://github.com/user-attachments/assets/74a3d64c-3db2-4604-9bd8-4b243d15cc4d)
